Library/OcBootManagementLib/OcBootManagementLib.c

```diff
@@ -241,21 +256,29 @@ OcRunBootPicker (
     //
     if (BootContext == NULL) {
       //
-      // TODO: Failed protocol hotkey can access OcPickerShowPicker mode even if
-      // this is denied by InternalRunRequestPrivilege above, is this OK?
+      // Because of this fallback code, failed protocol hotkey can access OcPickerShowPicker mode
+      // even if this is denied by InternalRunRequestPrivilege above. Believed to be acceptable
+      // as in a secure system any entry protocol drivers should be locked down and trusted.
       //
-      if (Context->HideAuxiliary || (Context->PickerCommand == OcPickerProtocolHotKey)) {
-        DEBUG ((DEBUG_INFO, "OCB: System has no boot entries, showing picker with auxiliary\n"));
+      if (Context->HideAuxiliary || (Context->PickerCommand == OcPickerProtocolHotKey) || IsApplePickerSelection) {
         Context->PickerCommand = OcPickerShowPicker;
         Context->HideAuxiliary = FALSE;
+        if (IsApplePickerSelection) {
+          DEBUG ((DEBUG_WARN, "OCB: Apple Picker returned no entry valid under OC, falling back to builtin\n"));
+          Context->PickerMode = OcPickerModeBuiltin;
+          gST->ConOut->ClearScreen (gST->ConOut);
```

Comment:
Apologies, but after putting up a hopefully quite useful PR, I am going to ask a really stupid question! This ClearScreen does not seem to do anything - i.e. the builtin picker text comes up over the still present final Apple picker screen - even though the ClearScreen not far below it definitely does do something to a graphics mode screen (hence the recent change to not call it when in graphics mode). What am I missing, if anyone can save me some time by pointing out the no-doubt obvious?